design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,271 @@
+4. The wizard asks the users if they want to import existing data to lakeFS. The user skips this step using the skip button.
+5. The wizard asks the user: “How are you using Spark”, and offers three alternatives: Databricks, AWS EMR, and spark-submit. The user chooses ‘Databricks’. It also asks the user for their lakeFS endpoint (and will show a default placeholder pointing to the current URL). The user types an **empty endpoint**.
+6. The wizard sends a GET request to the templating service with a query string of the format:
+    ```
+    ?template_location=s3://org-super-bucket/spark/databricksConfig.props
+    ```
+7. The templating service will fail to satisfy the `lakefs_url` template property and will return **‘400 Bad Request: error code 1’** to the wizard.
+8. The wizard will prompt a message saying that some needed information were not specified and that he should make sure he typed everything along the way.
+9. Continue with the flow as described above…
+
+### Sad flow - No bucket permissions
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to enter the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user skips this step using the skip button.
+5. The wizard asks the user: “How are you using Spark”, and offers three alternatives: Databricks, AWS EMR, and spark-submit. The user chooses ‘Databricks’. It also asks the user for their lakeFS endpoint (and will show a default placeholder pointing to the current URL). The user types the endpoint.
+6. The wizard sends a GET request to the templating service with a query string of the format:
+    ```
+    ?lakefs_url=<url>&template_location=s3://org-super-bucket/spark/databricksConfig.props
+    ```
+7. The templating service tries to fetch the template from the provided location and will fail to do so as the server doesn’t have sufficient permissions.
+8. The templating service will return **‘500 Internal Server Error: error code 1’** to the wizard.
+9. The wizard will prompt a message saying that the server could not access the bucket.
+10. Continue with the flow as described above…
+
+---
+
+## Monitoring
+
+### Metrics
+
+1. Templating service was called
+    ```json
+    {
+        "class": "templating_service",
+        "name": "calling_service",
+        "value": "<service name>"
+    }
+    ```
+2. Templating service- status 200
+    ```json
+    {
+        "class": "templating_service",
+        "name": "successful_call",
+        "value": "<service name>"
+    }
+    ```
+3. Templating service - status 500 - no access to provided template location
+    ```json
+    {
+        "class": "templating_service",
+        "name": "no_access",
+        "value": "<service name>"
+    }
```

Comment:
Don't we send status on our monitoring outputs as field values, rather than as metric names?